Role에 따른 권한 인증 기능

Habist opened the pull request:
## 작업 내용
컨트롤러의 요구 권한에 따라 클라이언트 요청을 처리할 수 있도록 변경하였습니다.

## 주의 사항

>  참조이슈
> #42

Habist committed:
refactor: [#42] Request URL 오타 수정

관리자 요청 처리를 담당하는 URL을 수정했습니다.

Habist committed:
feat: [#42] Role에 따른 권한 인증 기능 개발

클라이언트 요청을 필요 권한에 따라 처리할 수 있도록 추가했습니다.
권한 Enum 클래스의 순서를 정할 수 있도록 정수 값을 추가했습니다.
사용자의 권한을 반환하는 메소드를 추가했습니다.

Habist committed:
refactor: [#42] 관리자 컨트롤러 권한 인증 추가

관리자 요청을 담당하는 컨트롤러에 권한을 확인하도록 변경하였습니다.

f-lab-dev commented on src/main/java/com/campool/annotation/LoginValidation.java:
```diff
@@ -9,4 +10,6 @@
 @Retention(RetentionPolicy.RUNTIME)
 public @interface LoginValidation {
 
+    Role authorization() default Role.USER;
```
이름을 그냥 role이라고 해도 충분히 의미가 전달되지 않을까요?

Habist replied:
`role`이라고 하면 뭔가 역할이라는 **좁은 의미**에 국한되어 해당 `role`만 처리할 수 있는 의미로 느껴졌고
`authorization`은 요청을 처리할 수 있는 `role`이 하위 `role`까지 **포함하는 의미**로 느껴졌습니다.

누군가 봤을 때 role 만으로 의미가 충분히 전달될 수 있다면 role도 괜찮다고 생각합니다!

리뷰 반영할게요~

Habist committed:
refactor: [#42] 로그인 검증 애노테이션 속성, 변수 명칭 변경

LoginValidation의 권한 속성 명칭을 role로 변경하였습니다.
로그인 권한 처리 로직에서 role의 order를 사용하는 코드의 변수명을 더 명확한 의미로 변경하였습니다.
